internal/groups/api/transport.go

```diff
@@ -190,21 +191,21 @@ func readStringQuery(r *http.Request, key string) (string, error) {
 }
 
 func readMetadataQuery(r *http.Request, key string) (map[string]interface{}, error) {
-	vals := bone.GetQuery(r, key)
-	if len(vals) > 1 {
+	q, err := url.ParseQuery(r.URL.RawQuery)
```

Comment:
What seems to be a problem with `bone`'s query and why are we switching to Go's stdlib `url`?

Reply:
As @mteodor explained [here](https://github.com/mainflux/mainflux/issues/1357#issuecomment-774448180), Bone is not aware that the provided value is JSON and it's splitting the underlying string by `,` creating an array ([here is the code](https://github.com/go-zoo/bone/blob/31c3a0bb520c6d7a63dbb942459a3067787a975e/helper.go#L148)) instead of returningng a single string that contains JSON query.

Reply:
Before we go and use Go stdlib, I would like that we consider 2 things:
1. Open an issue to Bone
2. Examine how [Chi](https://github.com/go-chi/chi) is handling this, and maybe implement https://github.com/mainflux/mainflux/issues/963 first

Reply:
1. I would say no since this is by design (and will potentially cause a lot of code running Bone to break).
2. We should examine this multiplexer (it looks promising) but I would say that it's a separate issue while this PR is only a fix for the reported bug.

Reply:
Agree with @dusanb94 , this PR resolves the current bug. Let's fix it ASAP, by merging a fix. Then consider #963 which is a big change.